packages/x-data-grid-premium/src/DataGridPremium/DataGridPremium.tsx

```diff
@@ -593,6 +593,14 @@ DataGridPremiumRaw.propTypes = {
    * @param {string} data The data copied to the clipboard.
    */
   onClipboardCopy: PropTypes.func,
+  /**
+   * Callback fired before the clipboard paste operation starts.
+   * Use it to confirm or cancel the paste operation.
+   * @param {object} params Params passed to the callback.
+   * @param {string[][]} params.data The raw pasted data split by rows and cells.
+   * @returns {Promise<any> | void} A promise that resolves to confirm the paste operation, and rejects to cancel it.
```

Comment:
additionally to my comment above: Here the description state that it returns a Promise the resolves or rejects, but the signature shows `Promise | void` as return value